src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -25,7 +29,9 @@ internal class MvcEndpointDataSource : EndpointDataSource
         public MvcEndpointDataSource(
             IActionDescriptorCollectionProvider actions,
             MvcEndpointInvokerFactory invokerFactory,
-            IEnumerable<IActionDescriptorChangeProvider> actionDescriptorChangeProviders)
+            IEnumerable<IActionDescriptorChangeProvider> actionDescriptorChangeProviders,
+            IOptions<MvcEndpointDataSourceOptions> options,
```

Comment:
heh, I was in the middle of doing this for conventional routing. You beat me to it :)

Reply:
This was just a hack, I like the one you are going to do :-)